packages/lit-html/src/directive-helpers.ts

```diff
@@ -118,12 +118,19 @@ export const insertPart = (
     const endNode = wrap(part._$endNode!).nextSibling;
     const parentChanged = part._$parent !== containerPart;
     if (parentChanged) {
+      const newConnectionState = containerPart._$isConnected;
+      const shouldNotifyConnectionChanged =
+        part._$notifyConnectionChanged !== undefined &&
+        newConnectionState !== part._$isConnected;
```

Comment:
Do these have to be computed before the next line? I'd move them closer to the `part._$notifyConnectionChanged` call if possible.

Reply:
`part._$isConnected` relies on the part's current parent, which is changed on the next line.

I refactored it a bit, PTAL

Reply:
looks good